app/Http/Requests/User/DeleteUserRequest.php

```diff
@@ -21,11 +20,7 @@ class DeleteUserRequest extends BaseApiRequest implements HasUser
 	 */
 	public function authorize(): bool
 	{
-		// This should always return true, because we already check that the
-		// request is made by an admin during authentication (see
-		// `routes/web.php`).
-		// But better safe than sorry.
-		return Gate::check(UserPolicy::IS_ADMIN);
+		return Gate::check(UserPolicy::CAN_CREATE_OR_EDIT_OR_DELETE, User::class);
```

Comment:
Why that? As far as I can tell from code inspection, this is identical to the method of the parent class.

I assumed the whole point why you moved most of the code from `AbstractUserRequest` into a new `AbstractUsernamePasswordRequest` and therewith created another layer of indirection was that you could inherit this class from `AbstractUserRequest` (without pulling in the password-related parts) and then re-use the `authorize` method here.

But if you don't do that I wonder why you did the first step at all.

Reply:
Refactored.